web/src/app/administration/components/workbasket-distribution-targets/workbasket-distribution-targets.component.spec.ts

```diff
@@ -4,60 +4,118 @@ import { of } from 'rxjs';
 import { SavingWorkbasketService } from 'app/administration/services/saving-workbaskets.service';
 import { RequestInProgressService } from 'app/shared/services/request-in-progress/request-in-progress.service';
 
+import { LinksWorkbasketSummary } from 'app/shared/models/links-workbasket-summary';
 import { configureTests } from 'app/app.test.configuration';
 import { InfiniteScrollModule } from 'ngx-infinite-scroll';
-import { Side,
-  WorkbasketDistributionTargetsComponent } from './workbasket-distribution-targets.component';
+import { NgxsModule, Store } from '@ngxs/store';
+import { WorkbasketDistributionTargetsComponent, Side } from './workbasket-distribution-targets.component';
 import { WorkbasketDualListComponent } from '../workbasket-dual-list/workbasket-dual-list.component';
-import { NotificationService } from '../../../shared/services/notifications/notification.service';
+import { NotificationService } from '../../../shared/services/notifications/notification.service'; import { ClassificationSelectors } from '../../../shared/store/classification-store/classification.selectors';
+import { WorkbasketSelectors } from '../../../shared/store/workbasket-store/workbasket.selectors';
 
 describe('WorkbasketDistributionTargetsComponent', () => {
   let component: WorkbasketDistributionTargetsComponent;
   let fixture: ComponentFixture<WorkbasketDistributionTargetsComponent>;
   let workbasketService;
-  const workbasket = new Workbasket('1', '', '', '', ICONTYPES.TOPIC, '', '', '', '', '', '', '', '', '', '', '', '',
-    { distributionTargets: { href: 'someurl' } });
+  const workbasket = createWorkbasket('1', '', '', '', ICONTYPES.TOPIC, '', '', '', '', '', '', '', '', '', '', '', '',
+    {});
+
+  function createWorkbasket(workbasketId?, created?, key?, domain?, type?, modified?, name?, description?,
+    owner?, custom1?, custom2?, custom3?, custom4?, orgLevel1?, orgLevel2?, orgLevel3?, orgLevel4?,
+    _links?: Links, markedForDeletion?: boolean): Workbasket {
+    return {
+      workbasketId,
+      created,
+      key,
+      domain,
+      type,
+      modified,
+      name,
+      description,
+      owner,
+      custom1,
+      custom2,
+      custom3,
+      custom4,
+      orgLevel1,
+      orgLevel2,
+      orgLevel3,
+      orgLevel4,
+      markedForDeletion,
+      _links
+    };
+  }
+
+  function createWorkbasketSummary(workbasketId, key, name, domain, type, description, owner, custom1, custom2, custom3, custom4) {
+    const workbasketSummary: WorkbasketSummary = {
+      workbasketId,
+      key,
+      name,
+      domain,
+      type,
+      description,
+      owner,
+      custom1,
+      custom2,
+      custom3,
+      custom4
+    };
+    return workbasketSummary;
+  }
+
+  const workbasketSummaryResource: WorkbasketSummaryRepresentation = {
+    workbaskets: [
+      createWorkbasketSummary('1', 'key1', 'NAME1', '', 'PERSONAL',
```

Comment:
again, i do not see the benefit of a function just returning an object, when we could just create the object without problem. This function does not even increase readability, as i will not know which attribute gets assigned a '1' or a ''.

Maybe i am missing something